add gdsfactory

joamatab opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x]  Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * The top level meta keys are in an unexpected order. Expecting ['package', 'source', 'build', 'requirements', 'test', 'about', 'extra'].
 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found it was in an excellent condition.

joamatab commented:
Ready for review!

@conda-frorge/staged-recipes

joamatab commented:
Ready for review 

@conda-forge/staged-recipes

ocefpaf commented:
@joamatab this is not passing and it is not ready for review. It seems you need to also submit a PR to add https://pypi.org/project/lytest/, which is a dependency here.

joamatab commented:
Thank you Filipe!

I updated gdsfactory where i added lytest as an optional dependency

the recipe is ready for review now

@ocefpaf 
@conda-forge/staged-recipes

joamatab commented:
ready for review

ocefpaf commented:
Looks like there is an entrypoint: `gf=gdsfactory.gf:gf` Maybe we can make this one noarch if we specify that in the meta.yaml.

joamatab commented:
thank you Filipe! 

i added the entrypoint

@ocefpaf 
@conda-forge/staged-recipes

BastianZim commented on recipes/gdsfactory/LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
The license is already included in the tarball so it can be removed.

joamatab replied:
fixed in https://github.com/conda-forge/staged-recipes/pull/19346/commits/de9027840e81d014190a625b259452dcbf7d6167

BastianZim commented on recipes/gdsfactory/meta.yaml:
```diff
@@ -0,0 +1,66 @@
+  skip: true  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - gf = gdsfactory.gf:gf
```
You have entry points and scripts upstream. Are both of these needed?

Ref:

Scripts: https://github.com/gdsfactory/gdsfactory/blob/5f5b045b442b372241ba2c2158e296f7f72afd11/setup.py#L37
Entry points: https://github.com/gdsfactory/gdsfactory/blob/5f5b045b442b372241ba2c2158e296f7f72afd11/setup.py#L46

Only entry points would be better as the package can then be noarch.

ocefpaf replied:
@joamatab did you try this suggestion? noarch packages are better b/c require less rebuilds.

joamatab replied:
yes Filipe, i added entry points as suggested by @BastianZim 

See https://github.com/conda-forge/staged-recipes/pull/19346/files

ocefpaf replied:
> @joamatab did you try this suggestion? noarch packages are better b/c require less rebuilds.

@joamatab we are talking about making it `noarch` now that you have the entrypoints. You will need a with `noarch: python` above the entrypoint.

joamatab replied:
Hi Filipe,

just added `noarch: python`

https://github.com/conda-forge/staged-recipes/pull/19346/commits/cfe2b218e4086f0dcaa866d609391d6bccf06926

BastianZim reviewed (changes requested):
Just a couple of small things, then this is ready to go 🚀

joamatab commented:
thank you Bastian

I fixed the license and only keep the entry points

Ready for review

@BastianZim 
@conda-forge/staged-recipes

joamatab commented:
Ready for review

joamatab commented:
Ready for review 

@conda-forge/staged-recipes

joamatab commented:
@proppy

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.
 * noarch: python recipes are required to have a lower bound on the python version. Typically this means putting `python >=3.6` in **both** `host` and `run` but you should check upstream for the package's Python compatibility.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

BastianZim commented:
Please also follow the suggestion of the linter and remove the skip part.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found it was in an excellent condition.

joamatab commented:
Perfect 

Thank you @BastianZim and @ocefpaf 

Ready for review
